lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -131,4 +131,19 @@ def error_message(response)
       JSON.parse(response.body)['error']['full_messages'].join(' ') rescue "Response: #{response.code} #{response.message}"
     end
   end
+
+  def search(id_name, value)
+    if !value.nil? and value.start_with?("search=") then
+      lookup_uri = "api/v2/" + id_name + "?" + value
+      lookup = request(:get, lookup_uri)
```

Comment:
I think this creates some redundancy. We already have `request` which accepts a `params` hash so if anything this should pass search as a hash. That properly escapes it.

Reply:
@ekohl is there an example of using `request` I could have a look at to test with and make sure this works with it?

Reply:
There are some examples of using #request in its spec coverage: https://github.com/theforeman/puppet-foreman/blob/master/spec/unit/foreman_resource_rest_v3_spec.rb#L64-L131